qcodes/dataset/dond/do_nd.py

```diff
@@ -34,104 +35,438 @@
+        self._write_period = write_period
+        self._log_info = log_info
+        self._dataset_dependencies = dataset_dependencies
+        self._shapes = self._get_shape()
+        self._groups = self._create_groups()
+
+        if log_info is not None:
+            self._extra_log_info = log_info
+        else:
+            self._extra_log_info = "Using 'qcodes.dataset.dond'"
+
+    def _get_shape(self) -> Shapes | None:
+        try:
+            shapes: Shapes | None = detect_shape_of_measurement(
+                self._measurements.measured_parameters, self._sweeper.shape
+            )
+            LOG.debug("Detected shapes to be %s", shapes)
+        except TypeError:
+            LOG.exception(
+                f"Could not detect shape of {self._measurements.measured_parameters} "
+                f"falling back to unknown shape."
+            )
+            shapes = None
+        return shapes
+
+    def _get_experiments(
+        self, experiments: Experiment | Sequence[Experiment] | None
+    ) -> Sequence[Experiment | None]:
+        if not isinstance(experiments, Sequence):
+            experiments_internal: Sequence[Experiment | None] = [
+                experiments for _ in self._measurements.grouped_parameters
+            ]
+        else:
+            experiments_internal = experiments
+
+        if len(experiments_internal) != len(self._measurements.grouped_parameters):
+            raise ValueError(
+                f"Inconsistent number of "
+                f"parameter groups and experiments "
+                f"got {len(self._measurements.grouped_parameters)} and {len(experiments_internal)}"
+            )
+        return experiments_internal
+
+    def _create_groups(self) -> tuple[_SweapMeasGroup, ...]:
+
+        if self._dataset_dependencies is None:
+            setpoint = self._sweeper.all_setpoint_params
+            meaure_groups = self._measurements.grouped_parameters
+
+            groups = []
+            sp_group: Sequence[ParameterBase]
+            m_group: Sequence[ParamMeasT]
+            for m_group, experiment, meas_name in zip(
+                meaure_groups,
+                self._experiments,
+                self._measurements.measurement_names,
+            ):
+                meas_ctx = self._create_measurement_cx_manager(
+                    experiment, meas_name, setpoint, tuple(m_group)
+                )
+                s_m_group = _SweapMeasGroup(
+                    setpoint, tuple(m_group), experiment, meas_ctx
+                )
+                groups.append(s_m_group)
+        else:
+            potential_setpoint_groups = self._sweeper.sweep_groupes
+            requested_measure_groups = self._measurements.grouped_parameters
+
+            if len(self._dataset_dependencies) != len(requested_measure_groups):
+                raise ValueError(
+                    f"Requested for data to be split into {len(requested_measure_groups)} "
```

Comment:
perhaps? or is "groups" clear enough?
```suggestion
                    f"Requested for data to be split into {len(requested_measure_groups)} datasets"
```